### DISCUSSION TITLE
no lib folder when compiling arnold-usd

### DISCUSSION THREAD

**Discussion by raphaeljnwave** [Autodesk/arnold-usd#1257] [Q&A]
Hello Arnold team !

I am a little bit confused with the installation part from the documentation.
Once compiled, I have a LICENSE.md file and bin/docs/include/plugin/procedural/schema folder.

However, the installations is talking about
<arnold-usd_dir>/lib/python
<arnold-usd_dir>/lib/usd
<arnold-usd_dir>/lib

Can you tell me more about those folders ? Did I failed my compilation ?
I tried several things but I always have the same results.

Thank you !

**Comment by cpichard** [comment]
Hi @raphaeljnwave ,
did you compile with cmake or abuild ?
C

**Comment by raphaeljnwave** [comment]
Hi @cpichard,

Thank you for answering. I tried both cmake and abuild. here are my abuild settings:

All dependencies are downloaded and compiled in the 'Downloads' folder. Here is my custom.py located next to the abuild script:

```
ARNOLD_PATH='/home/raphaelj/Downloads/arnold-7.1.2.0-sdk'

USD_PATH='/home/raphaelj/Downloads/USD-21.11/install'
USD_BIN='/home/raphaelj/Downloads/USD-21.11/install/bin'
USD_INCLUDE='/home/raphaelj/Downloads/USD-21.11/install/include'
USD_LIB='/home/raphaelj/Downloads/USD-21.11/install/lib'
USD_LIB_PREFIX='libusd_'

PREFIX='/home/raphaelj/Downloads/arnold-usd-Arnold-7.1.2.0/install'
PREFIX_LIB='/home/raphaelj/Downloads/arnold-usd-Arnold-7.1.2.0/install/lib'

BOOST_INCLUDE='/home/raphaelj/Downloads/boost_1_70_0/install/include/boost'
BOOST_LIB='/home/raphaelj/Downloads/boost_1_70_0/install/lib'
BOOST_LIB_NAME='libboost_%s37'

TBB_INCLUDE='/home/raphaelj/Downloads/oneTBB-2018_U6/include/tbb'
TBB_LIB='/home/raphaelj/Downloads/oneTBB-2018_U6/install/release'

PYTHON_INCLUDE='/home/raphaelj/Downloads/Python-3.7.7/install/include/python3.7m'
PYTHON_LIB='/home/raphaelj/Downloads/Python-3.7.7/install/lib'
PYTHON_LIB_NAME='python3.7m'

USD_BUILD_MODE='shared_libs'

SHCXX='/opt/rh/devtoolset-6/root/usr/bin/g++'
CXX_STANDARD='14'
DISABLE_CXX11_ABI=True

BUILD_SCHEMAS=True
BUILD_RENDER_DELEGATE=True
BUILD_NDR_PLUGIN=True
BUILD_PROCEDURAL=True
BUILD_TESTSUITE=True
BUILD_USD_WRITER=True
BUILD_DOCS=False
```
I then call `./abuild -j 12 install` from the arnold usd root folder.

Thanks for your help !
Raphael

**Reply by cpichard** [reply]
Hi @raphaeljnwave,
Thanks for your reply.
I think the documentation might be out of date, I'll double check,
Cyril

**Reply by raphaeljnwave** [reply]
Thanks @cpichard !

Let me now if you have some info :)
Raphael

**Reply by Lucas3Dspain** [reply]
I got Arnold working on usdView last week.
I compiled with cmake. I could share my process and binaries if that helps you.

**Reply by raphaeljnwave** [reply]
Hi @Lucas3Dspainm,

It would be great, thanks ! So you have a lib folder after compiling it, right ? i guess I missed a step.

**Reply by Lucas3Dspain** [reply]
As @sebastienblor said, there is no lib folder.
note: maybe the DBUILD_SCHEMAS flag has something to do with it.

But if this help you in any way, this is the command i used:

`"C:\Program Files\CMake\bin\cmake.exe" ^ -G "Visual Studio 16 2019" ^ -DCMAKE_INSTALL_PREFIX="G:\USD_builds\my_builds\USD_arnold_sandbox" ^ -DARNOLD_LOCATION="C:\Program Files\Autodesk\Arnold\Arnold-7.1.3.0-windows" ^ -DUSD_INCLUDE_DIR="G:\USD_builds\my_builds\21_08\USD_complete\include" ^ -DUSD_LIBRARY_DIR="G:\USD_builds\my_builds\21_08\USD_complete\lib" ^ -DBUILD_SCHEMAS=OFF ^ -DTBB_FOUND=ON ^ -DTBB_INCLUDE_DIRS="G:\USD_builds\my_builds\21_08\USD_complete\include" ^ -DTBB_LIBRARIES="G:\USD_builds\my_builds\21_08\USD_complete\lib\tbb.lib" ^ -DBUILD_USE_CUSTOM_BOOST=ON ^ -DBoost_INCLUDE_DIRS="G:\USD_builds\my_builds\21_08\USD_complete\include\boost-1_70" ^ -DBoost_LIBRARIES="G:\USD_builds\my_builds\21_08\USD_complete\lib\boost_python39-vc142-mt-x64-1_70.lib" ^ -DBUILD_USE_PYTHON3=ON ^ -DPython3_ROOT_DIR="C:\Users\LucasMorante\AppData\Local\Programs\Python\Python39" ^ "C:\Users\LucasMorante\Desktop\_Repos\arnold-usd"`

and then 

`"C:\Program Files\CMake\bin\cmake.exe" --build . --config Release --target install`

**Comment by sebastienblor** [comment]
Hi @raphaeljnwave , yes this documentation is outdated. There's nothing being dumped in the lib folder anymore. 
In the past we were compiling the schemas as C++ libs, but now all the schemas info needed by the applications is stored in a simple .usda file, that should dumped in your builds.

What do you need to get exactly ? We have a ticket to restore the compilation of C++ schemas in #1198
